### PR TITLE
Feature/hardware endpoint

### PR DESCRIPTION
**Pull request by yash1337**
CRUD operations on the hardware endpoint. 
Added the DELETE endpoint which was missing in the old API.
Added tests for the endpoint.

### CODE REVIEW

**Comment by nathankurt** [comment]
reviewing

File: app.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from sqlalchemy.ext.automap import automap_base
 from sqlalchemy import create_engine
 from sqlalchemy.orm import Session
-from common.utils import unauthorised,headers
+from common.utils import unauthorised,headers,not_found
```
Comment (nathankurt) [inline app.py]: these should normally be on separate lines. so make it
```python
from common.utils import unauthorised
from common.utils import headers
from common.utils import not_found
```

Comment (nathankurt) [reply app.py]: also fix spelling of unauthorized, we wan't American English to be the standard here

File: app.py [hunk 2]
```diff
@@ -22,42 +22,68 @@
...
+api.add_resource(Faqs_CR,"/faqs")
+api.add_resource(Announcements_RUD,"/announcements/<int:announcement_id>")
+api.add_resource(Announcements_CR,"/announcements")
+api.add_resource(Hardware_RUD,"/hardware/<int:hardware_id>")
+api.add_resource(Hardware_CR,"/hardware")
 
-@app.route("/")#for flask app test and general info about the product
+@app.route("/")
 def helloworld():
+    """
```
Comment (nathankurt) [inline app.py]: Yo, put comments for a function above the function not inside of it.

File: common/json_schema.py [hunk 3]
```diff
@@ -1,14 +1,35 @@
 from marshmallow import Schema,fields
 
-class Faq_Schema(Schema):
-    """
+"""
     schema.dump = used for converting the automap.faqs object to a dictionary good for returning ie cleaning unncessary fields
-    schema.validate(request.get_json(force=True)) = used to validate if all the data required for updating and creating the faq is present.
-    """
+    schema.validate(request.get_json(force=True)) = used to validate if all the data required for updating and creating the faq is present.4
```
Comment (nathankurt) [inline common/json_schema.py]: Are these actually used in any of the files? I'm not seeing them in this file at all?

File: resources/announcements.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
+from flask_restful import Resource
+from werkzeug.exceptions import BadRequest
+from flask import request,jsonify,g
```
Comment (nathankurt) [inline resources/announcements.py]: throw these on separate lines 
```python
from flask import request
from flask import jsonify
from flask import g
```

File: resources/announcements.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        user_status,user = has_admin_privileges()
+        if user_status == "no_auth_token":
+            return (bad_request,400,headers)
+
+        if user_status == "not_logged_in":
+            return (unauthorised,401,headers)
+
+        if user_status == True:
+            try:
+                g.session.query(g.Base.classes.announcements).filter(g.Base.classes.announcements.id == announcement_id).one()
```
Comment (nathankurt) [inline resources/announcements.py]: What does this do? maybe throw something in there that explains

File: resources/announcements.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    def delete(self,announcement_id):
+        """
+        For DELETE request for specific id
+        """
+        user_status,user = has_admin_privileges()
+        if user_status == "no_auth_token":
+            return (bad_request,400,headers)
+
+        if user_status == "not_logged_in":
+            return (unauthorised,401,headers)
```
Comment (nathankurt) [inline resources/announcements.py]: fix unauthorized spelling

File: resources/announcements.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+        if user_status == True:
+            Announcements = g.Base.classes.announcements
+            try:
+                new_announcement = Announcements(
+                                                    title = data["title"],
+                                                    description = data["description"],
+                                                    pinned = data["pinned"],
+                                                    updated_at = datetime.now(),
```
Comment (nathankurt) [inline resources/announcements.py]: Think about using pendulum library for datetime stuff https://pendulum.eustace.io/

File: app.py (hunk 1 above)
Comment (yash1337) [reply app.py]: PEP 8 says to put them in the same line - https://www.python.org/dev/peps/pep-0008/?

I can fix the spellings!

File: common/json_schema.py (hunk 3 above)
Comment (yash1337) [reply common/json_schema.py]: They are used to validate the data given in the request. Check PUT or POST request handlers for any endpoint.

File: resources/announcements.py (hunk 5 above)
Comment (yash1337) [reply resources/announcements.py]: Okay!

File: resources/announcements.py (hunk 7 above)
Comment (yash1337) [reply resources/announcements.py]: The pendulum library doesn't add anything specific that is not being solved the inbuilt datetime library. We should try to keep the memory footprint of the API to a minimum.
https://aboutsimon.com/blog/2016/08/04/datetime-vs-Arrow-vs-Pendulum-vs-Delorean-vs-udatetime.html

Comment (nathankurt) [reply resources/announcements.py]: That's a good point, I agree.

File: app.py (hunk 1 above)
Comment (nathankurt) [reply app.py]: Oh yeah i guess if they are all coming from the same thing it's okay. Alright that's fine

File: resources/announcements.py (hunk 4 above)
Comment (nathankurt) [reply resources/announcements.py]: nvm you're good here

**Pull request merged**